### ISSUE TITLE
Suggestion: Better keyWest and keyEast in VueExcelFilter

### ISSUE THREAD

**Issue by ste-gr** [cscan/vue-excel-editor#88]
Hi,
I think there is a better solution for `keyWest` and `keyEast` in [VueExcelFilter.vue](src/VueExcelFilter.vue). Now there is a problem, when more than one column hidden in a row.
What do you think about this change:
```
keyWest(e) {
  var sel = document.getSelection();
  if (e.target.textContent === sel.toString() || sel.focusOffset === 0) {
    this.focusNext(e.target, el => el.previousElementSibling);
  }
},
keyEast(e) {
  var sel = document.getSelection();
  if (e.target.textContent === sel.toString() || sel.focusOffset >= e.target.textContent.length) {
    this.focusNext(e.target, el => el.nextElementSibling);
  }
},
focusNext(el, next) {
  var td = next(el);
  while (td && (td.tagName !== 'TD' || (td.style && td.style.display === 'none'))) {
    td = next(td);
  }
  if (td && td.focus) {
    td.focus();
  }
},
```
Greetings
Stefan

**Comment by cscan** [comment]
I see that moveEast and moveWest already consider the invisible flag. Any problem if you use them?

**Comment by ste-gr** [comment]
You mean, like `this.$parent.moveEast/West()`? I tried this, but it didn't work.
Maybe with some redesign, it's possible to work with callback/event at VueExcelEditor (I think also, it would be nicer).
As I saw in my suggestion is, that the first column isn't consider. So I adapted it a little:
```
keyWest() {
  var sel = document.getSelection();
  if (this.cell.textContent === sel.toString() || sel.focusOffset === 0) {
    this.focusNext(el => el.previousElementSibling, pos => pos - 1);
  }
},
keyEast() {
  var sel = document.getSelection();
  if (this.cell.textContent === sel.toString() || sel.focusOffset >= this.cell.textContent.length) {
    this.focusNext(el => el.nextElementSibling, pos => pos + 1);
  }
},
focusNext(next, nextPos) {
  var td = next(this.cell);
  var pos = nextPos(this.colPos);
  while (pos >= 0 && td && (td.tagName !== 'TD' || (td.style && td.style.display === 'none'))) {
    td = next(td);
    pos = nextPos(pos);
  }
  if (pos >= 0 && td && td.focus) {
    td.focus();
  }
},
```

**Comment by cscan** [comment]
Sorry I am wrong. I thought you are talking about the editor but filter, let me re-consider this.

**Comment by cscan** [comment]
Fixed, see v1.4.2
